pages/en/lb3/Defining-middleware.md

```diff
@@ -690,17 +690,10 @@ Instead, it will be ignored because the route was already matched.
 
 ### Error-handling middleware
 
-Use error-handling middleware to deal with request errors.
+Use error-handling middleware to deal with request errors. The  [`strong-error-handler`](https://github.com/strongloop/strong-error-handler/) middleware is scaffolded by default and sufficient for many projects, but additional custom error-handling middleware may be registered as necessary.
 While you are free to register any number of error-handling middleware, be sure to register them in the "final" phase.
-LoopBack registers two error-handling middleware by default:
 
-* `urlNotFound` middleware converts all requests that reach the middleware into an Error object with status 404,
-  so that 404 error responses are consistent with "usual" error responses.
```

Comment:
I think the `urlNotFound` middleware is different from the error handler. The reason why the REST adapters adds its own urlNotFound middleware is to prevent the top-level loopback app to add routes into the route space managed by strong-remoting.

Consider the following example:

```js
var loopback = require('loopback');
var app = loopback();

app.dataSource('db', { connector: 'memory' });
app.model(loopback.User, { dataSource: 'db' });

app.use('/api', loopback.rest());
app.get('/api/Users/foobar', (req, res) => res.send({ hello: 'world' }));

app.listen(3000, () => console.log('http://localhost:3000/'));
```

Currently, `GET /api/Users/foobar` returns 404 error `Unknown "User" id "foobar".` With the built-in `urlNotFound` handler disabled, the custom route would be run instead.

Honestly, I cannot tell whether that's a good or a bad thing, considering that the following code would call the custom route too:

```js
app.get('/api/Users/foobar', (req, res) => res.send({ hello: 'world' }));
app.use('/api', loopback.rest());
```

I am proposing to move this discussion to a standalone issue, so that other LoopBack users can get involved. As for this pull request, I let's keep the mention of `urlNotFound` middleware as it is now.

Thoughts?

> Maybe it's a mistake for there to be two of these? Is the plan to extract urlNotFound, as was done with strong-error-handler? Should it perhaps even be part of strong-error-handler? Any plans here?

We don't have any plans for that, but now that you mentioned it, I think it may make sense to provide this useful middleware in a place where other users can reuse it and strong-error-handler looks like the perfect vehicle for that!